src/couch_views/src/couch_views_indexer.erl

```diff
@@ -87,7 +88,49 @@ init() ->
         design_opts => Mrst#mrst.design_opts
     },
 
-    update(Db, Mrst, State).
+    try
+        update(Db, Mrst, State)
+    catch
+        exit:normal ->
+            ok;
+        Error:Reason  ->
+            NewRetry = Retries + 1,
+            RetryLimit = retry_limit(),
+
+            if NewRetry < RetryLimit -> ok; true ->
+                    NewData = add_error(Error, Reason, Data),
+                    couch_jobs:finish(undefined, Job, NewData),
+                    exit(normal)
+            end,
+
+            DataErr = Data#{<<"retries">> := NewRetry},
+            StateErr = State#{job_data := DataErr, last_seq := <<"0">>},
+            report_progress(StateErr, update)
```

Comment:
Technically for the transaction too large case, it might not make sense to retry unless we bisect the limit, But that's probably a separate commit. (Original idea for this is from @davisp. And we already do a similar pattern in the replicator:  https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch_replicator/src/couch_replicator_worker.erl#L382-L390).

But in this PR it might make sense to have a `should_retry(Retries, retry_limit(), {Error, Reason})` , maybe  something like:

```
should_retry(_, _, {erlfdb_error, 2101}) ->
   false;

should_retry(Retries, RetryLimit, _} when Retries < RetryLimit ->
   true;

should_retry(_, _, _} ->
   false;
```